src/Components/Facility/FacilityCreate.tsx

```diff
@@ -479,19 +424,20 @@ export const FacilityCreate = (props: FacilityProps) => {
     console.log(state.form);
     if (validated) {
       setIsLoading(true);
-      const data = {
+      const data: FacilityModel = {
         facility_type: state.form.facility_type,
         name: state.form.name,
         district: state.form.district,
         state: state.form.state,
         address: state.form.address,
-        pincode: state.form.pincode,
         local_body: state.form.local_body,
         features: state.form.features,
         ward: state.form.ward,
-        kasp_empanelled: JSON.parse(state.form.kasp_empanelled),
-        latitude: state.form.latitude || null,
-        longitude: state.form.longitude || null,
+        pincode: state.form.pincode,
+        location: {
+          latitude: Number(state.form.latitude),
+          longitude: Number(state.form.longitude),
+        },
```

Comment:
Are you sure coordinated is a nested JSON and is correct? Have you tested it? Kindly test your PR before requesting review!

```suggestion
        latitude: state.form.latitude || null,
        longitude: state.form.longitude || null,
```